Jolt/Physics/StateRecorder.h

```diff
@@ -39,6 +87,15 @@ class JPH_EXPORT StateRecorderFilter
 
 	/// If the state of a specific contact should be saved
 	virtual bool		ShouldSaveContact([[maybe_unused]] const BodyID &inBody1, [[maybe_unused]] const BodyID &inBody2) const { return true; }
+
+	///@}
+	///@name Functions called during RestoreState
+	///@{
+
+	/// If the state of a specific contact should be restored
+	virtual bool		ShouldRestoreContact([[maybe_unused]] const BodyID &inBody1, [[maybe_unused]] const BodyID &inBody2) const { return true; }
```

Comment:
I made this function const on purpose: to be consistent with the other functions in the class and because I think the filter itself should be const. I would do something like:

```C++
class MyProcessedContacts
{
public:
	void StoreContact(const BodyID &inBody1, const BodyID &inBody2)
	{
		// ...
	}
};

class MyStateRecorderFilter : public StateRecorderFilter
{
	virtual bool ShouldRestoreContact(const BodyID &inBody1, const BodyID &inBody2) const override
	{
		mContacts->StoreContact(inBody1, inBody2);
		return true;
	}
	
	MyProcessedContacts *mContacts;
};
```

Reply:
I understand, but this approach makes the API for applying multiple buffers unclear. The reason I suggested adding NotifyContactsRestored was to improve the clarity of the filter interface without obscuring this functionality.

Additionally, why is the filter object being made const? Doing so forces me to write less clean code, as I would prefer to use an array to track changes. It also requires me to allocate objects or rely on unnecessary const_casts.

Unless there’s a specific reason for making it const, allowing it to be non-const would result in a much cleaner and more straightforward implementation on both sides.

Reply:
Hey @jrouwe, I’ve already integrated this feature into the game, and it works great! Feel free to make the filter non-const or keep it as const. For now, I’ll continue using what I’ve integrated.

Thanks a lot for this functionality!